tests/src/Unit/SocialApiTest.php

```diff
@@ -0,0 +1,560 @@
+   */
+
+  public $machine_name;
+  public $name;
+  public $library;
+  public $min_version;
+  public $max_version;
+  protected $config;
+  private $networkManager;
+  protected $form = array();
+  protected $namespaces;
+  protected $module_handler;
+  protected $cache_backend;
+  public $type = "";
+  public $container;
+  private $networkManagers;
+  protected $client;
+  protected $session;
+  protected $sessionPrefix;
+  protected $storage;
+  protected $typed_config;
+  protected $event_dispatcher;
+  //UserManager
+  protected $entityType;
+  protected $entity_type_manager;
+  protected $messenger;
+  protected $logger_factory;
+  protected $plugin_id;
+  protected $user_manager;
+  protected $data_handler;
+  protected $current_user;
+  protected $dataHandler;
+  protected $sessionKeys;
+  //Netowrk
+  protected $configuration;
+  protected $plugin_definition;
+  protected $sdk = 'drupal123';
+
+
+
+  /**
+   * __construct function
+   */
+  public function __construct() {
+       parent::__construct();
+   }
+
+
+  /**
+   * {@inheritdoc}
+   */
+
+  public function setUp() {
+    // enable any other required module
+    parent::setUp();
+    // $this->socialPostEntityDeleteForm = $this->getMock($SocialPostEntityDeleteForm::class, ['getRedirectUrl']);
+  }
+
+  /**
+   * test for class OAuth2Manager
+   */
+
+  public function testOAuth2Manager () {
+    // assertion to check if the file exists
+    // $this->assertFileExists('../drupal8/modules/social_api/src/AuthManager/OAuth2Manager.php');
+
+    $abstractClass = 'Drupal\social_api\AuthManager\OAuth2Manager';
+
+    //
+    // checking for correct getClient and setClient methods
+    //
+     $mockAbstractClass = $this->getMockBuilder($abstractClass)
+       // ->setMethods(array('setClient','getClient'))
+                               ->getMockForAbstractClass();
+       // $mock->expects($this->once())
+       //          ->method('setClient')
+       //          ->with($this->equalTo('something'));
+
+     $mockAbstractClass->setClient('12345');
+     // var_dump($mockAbstractClass->getClient());
+     $this->assertEquals('12345', $mockAbstractClass->getClient());
+
+     //
+     // checking for correct getAccessToken and getAccessToken methods
+     //
+      $mockAbstractClass = $this->getMockBuilder($abstractClass)
+        // ->setMethods(array('setAccessToken','getAccessToken'))
+                                ->getMockForAbstractClass();
+        // $mock->expects($this->once())
+        //          ->method('setClient')
+        //          ->with($this->equalTo('something'));
+
+      $mockAbstractClass->setAccessToken('12345');
+      // var_dump($mockAbstractClass->getAccessToken());
+      $this->assertEquals('12345', $mockAbstractClass->getAccessToken());
+
+
+
+      //check for functions if present
+      $this->assertTrue(
+        method_exists($mockAbstractClass, 'setClient'),
+          'OAuth2Manager does not have setClient function/method'
```

Comment:
I feel like does not *implement* sounds better and is technically the right word.

Reply:
Changes done.